flight-icons/CONTRIBUTING.md

```diff
@@ -90,36 +52,103 @@ where `###` is your personal access token. You can use the `.env.template` file
 
 **IMPORTANT**: the `.env` file is ignored by git, and should not be committed to the repository.
 
-### Running the scripts
 
-Finally, you can run the scripts commands via CLI:
+Once you've done the initial setup, there are three distinct steps in the process that you can follow.
+
+## Sync
+
+The synchronization step is executed by the `/scripts/sync.ts` file. To run this script launch the following commands in your CLI (while in the `flight-icons` folder):
 
 ```bash
-# sync
 yarn sync
+```
 
-# build
-yarn build
+This action will:
+
+* Retrieve the assets metadata from Figma via REST API
+* Export the assets as `.svg` files into `./svg-original/`
+* Generate the `catalog.json` file
+
+## Build
 
-# release
-[TODO]
+The build step is executed by the `/scripts/build.ts` file. To run this script use the following command in your CLI (while in the `flight-icons` folder):
+
+```bash
+yarn build
 ```
 
-## 🚧 [WIP] Releasing a new npm version of the package
+This action will:
+
+* Optimize the SVG files and save then in a `temp` folder
+* Process the optimized SVG files and save then in the `svg` folder
+* Update the existing files in the Ember addon folder:
+    * Process the optimized SVG, generate a SVG sprite, and overwrite the existing sprite
+    * Overwrite the catalog.json with the new one
+    
+_**Notice**: as mentioned above, this step not only updates the content of the `flight-icons` folder, but also the content of the `ember-flight-icons`. You will see these changes reflected in your git diff status._
+
+## Release
+
+The release step is executed using a set of NPM scripts defined in the `package.json` file. Use the following command in your CLI (while in the `flight-icons` folder):
+
 
 ```bash
-cd flight-icons
+yarn release
 ```
 
-- Bump the version number, per SemVer, for the `flight-icons/package.json`.
+_**IMPORTANT**: if you need to do some tests, use a **local** package registry (see below), don't test directly in production!_
+
+
+This action will:
```

Comment:
does this action also open a PR with the version number changes so that they are committed to the repo or is that a step that needs to be documented? If so, are there any risks to the idea that releases could happen npm that are not actually reflected in the package.json files here?

Reply:
No, you have to manually open the PR. You're totally right, we should add that step. 

We can do it the way we had it before (1) PR with version bump reviewed and merged to `main`, and then (2) `npm publish`, but arguably no need for the tool then. Add the step and then see how it goes with someone new trying publish out?

Reply:
Let me update and clarify this in the documentation, then we can discuss if it's clear enough.

Reply:
ok, I have checked but actually that's already documented at the top of the document ("Releasing a new npm version of the package"). do you think this should be repeated here? or highlighted in another way?

Reply:
is it [this line](https://github.com/hashicorp/flight/pull/229/files#diff-ed6c88899856b43588a6bf4f87b594d33966182a453b0e88d705cca1144c0d3bR15) that you're referring to?

Reply:
Yes, and the line below:

> The update process for the icons should happen in a dedicated branch, associated with the GitHub task/issue. Once you have run the `yarn sync` and `yarn build` (see the instructions below), check the diff in your git client, and once you're OK with them commit the changes to the branch, push and then ask for a pull request.
> 
> Once this request has been approved and the branch merged in `main`, you can publish the package (using directly the `main` branch, no need to create a new PR for this) using the `release` command (see the instructions below).

Do you think is a problem if someone merges directly in `main` the bumped version? Yesterday @amyrlam created a PR for the "bump" (https://github.com/hashicorp/flight/pull/227/files), but for me is not necessary: at that point the packages are already on npm with that version, what do you need to review?

@amyrlam @Dhaulagiri if you think the "flow" as described above doesn't... flow, write down in details the steps you expect to see/do, and I'll update the scripts and the documentation accordingly.

Reply:
I have clarified the text a little bit, hope now is clearer:

> The update process for the icons should happen in a dedicated branch, associated with the GitHub task/issue. Once you have run the `yarn sync` and `yarn build` (see the instructions below), check the diff in your git client, and once you're OK with them commit & push the changes to the branch, and then open a pull request in GitHub.
> 
> Once this request has been approved and the branch merged in `main`, you can publish the package (using directly the `main` branch, no need to create a new PR for this) using the `release` command (see the instructions below).
>

Reply:
It's definitely possible I'm misunderstanding the flow here, but the instructions for opening a PR are specific to sync/build and don't reference that _after_ you follow this release process the actual package.json version number changes would still need to be committed, no? If you didn't for some reason, someone else could come along after and run the release script and try to release a version that is already on npm?

> Do you think is a problem if someone merges directly in main the bumped version?

At least with our current branch protection policies this is not allowed, but this is often a problem with automating releases so happy to see if we can figure something out.

The flow I'm more familiar with is:

1. Update version numbers in package.json & any changelogs that accompany the release (this can either be manual or automated)
2. Open a PR with those changes
3. Someone else reviews to sanity check nothing was missed and that the version number matches the changes that occurred (i.e. nothing is a breaking change).
4. PR is merged
5. Tag new releases in Git and publish to npm

Reply:
Created a PR for splitting the "bump" and the "publish". 
https://github.com/hashicorp/flight/pull/236

Now I'll update the documentation.

Reply:
@Dhaulagiri @amyrlam see my latest update to the documentation. Let me know if now is ok, and clear enough.

> Tag new releases in Git
Re. the tags, at the moment I don't see any tag in GitHub (apart from a few ones that I created during my testing, which I have deleted). Are we supposed to tag the release too?

Reply:
I don't think we want to commit to `main`

FWIW I think https://github.com/release-it/release-it does the npm publish and the tagging, but not in the order Brian is suggesting, the PR comes after. If we want to go back that route, it seemed easy to set up if we publish both individually, like this PR (the tricky thing for me, was publishing them both in tandem from root)

Maybe the best way to publish is to have a buddy in a similar way to a deploy process, have to have someone there and "pair" on the checklist?

Reply:
> I don't think we want to commit to `main`

What do you mean? For tagging? I think is possible to create tags independently from the branch.

Re. the creation of tag, we can imagine to update our script command with `yarn publish --tag [tag-number-here]` or we can do something similar to what this guy has done: https://www.promptworks.com/blog/automated-package-publishing-with-github-actions

Reply:
also, I checked and `version-bump-prompt` can do all of them: bumping, committing, tagging and pushing 
https://www.npmjs.com/package/version-bump-prompt#git-tag
but as you say, this is not the order we want

Reply:
or we can go directly with a `git` command: https://git-scm.com/book/en/v2/Git-Basics-Tagging

Reply:
To clarify, I don't think the package.json version bumps should be committed to `main` directly. Realize this isn't production software per se but opening up `main` to direct commits could lead to others accidentally doing that. I accidentally did that on `waypoint` myself.

I've personally never found much of a benefit to git tags... but no strong opinion

We should probably have a CHANGELOG also for 1.0.

@didoo @Dhaulagiri I think it would make sense to merge this branch in and iterate, what do you think? We need to add a step about the PR, but it might be nice to just try it without actually publishing to `npm`. I'm happy to take another stab at the directions later today and @Dhaulagiri or @MelSumner could run the steps on Monday for a rc candidate in npm?

Reply:
👍

Reply:
I made https://github.com/hashicorp/flight/issues/240 and will merge this in